lib/pbench/server/cache_manager.py

```diff
@@ -491,12 +574,13 @@ def _get_metadata(tarball_path: Path) -> Optional[JSONOBJECT]:
         """
         name = Dataset.stem(tarball_path)
         try:
-            data = Tarball.extract(tarball_path, f"{name}/metadata.log")
-        except BadDirpath:
+            data: IO[bytes] = Tarball.extract(tarball_path, f"{name}/metadata.log")
```

Comment:
This was more for the IDE than for you, because it was left in the dark and became profoundly unhelpful at one point. 😆

Reply:
Poor IDE!  However, it really shouldn't have been in the dark, because it _should_ have known what `extract()` would return...unless that was somehow poorly-defined.

Reply:
The key was "at one point" ... while I was playing with signatures and hadn't actually changed the `Tarball.extract` type hint. This was helpful. It's redundant now.